examples/leasing/node_release_script.py

```diff
@@ -0,0 +1,241 @@
+    nodes_to_update_infile = list(Set(
+        non_persistent_list)-Set(free_node_list))
+
+    for node in nodes_to_update_infile:
+        # get the information from status file for node.
+        node_info_from_file = get_project_and_time_for_node(statusfile, node)
+        project_in_file = node_info_from_file[0]
+        old_time = node_info_from_file[1]
+
+        # get the information from HIL for node
+        node_current_info = hil_client.node.show(node)
+        project_in_hil = node_current_info['project']
+        # Increase the time by one unit
+        new_time = int(node_info_from_file[1])+1
+        # See if the node is in the same project and if it is outside
+        # free pool for time less than threshold value then update only
+        # time in status file
+        if (project_in_file == project_in_hil and
+                new_time < threshold_time):
+            update_time_for_node(statusfile, node, old_time, new_time)
+        # See if the node is in the same project and if it is outside
+        # free pool for time greater than threshold value then release
+        # the node back to free pool
+        elif (project_in_file == project_in_hil and
+                new_time >= threshold_time):
+            # Detaching nodes from networks
+            detach_networks_from_node(hil_client, node)
+            # Releasing node from project.
+            release_from_project(hil_client, statusfile, node, project_in_hil)
+
+        # See if the node is not in free pool and check if the projects
+        # are not matching then we need to update project in status file
+        elif (project_in_file != project_in_hil and
+                project_in_hil is not None):
+            update_project_in_status_file(
+                statusfile, node, project_in_hil,
+                project_in_file)
+
+
+def get_project_and_time_for_node(statusfile, node):
+    """Reads current information from file for every node."""
+
+    node_status = ''
+    node_project = ''
+    node_duration = 0
+    with open(statusfile, 'r') as status_file:
+        for line in status_file:
+            node_status = line.split()
+            if node == node_status[0]:
+                node_project = node_status[1]
+                node_duration = node_status[2]
+    status_file.close()
+    return (node_project, node_duration)
+
+
+def update_time_for_node(
+        statusfile, node,
+        old_time, new_time
+        ):
+    """Updates the time for a node in file"""
+
+    newline = ''
+    with open(statusfile, 'r') as status_file,\
+            open('/tmp/tempfile', 'w') as output_file:
+        for line in status_file:
+            words = line.split()
+            if node == words[0]:
+                words[2] = str(new_time)
+                newline = ' '.join(words)+"\n"
+                output_file.write(newline)
+            else:
+                output_file.write(line)
+    status_file.close()
+    output_file.close()
```

Comment:
same thing, you opened your files in a `with` block, so these lines are redundant.